fv3config/fv3run/_docker.py

```diff
@@ -35,59 +36,38 @@ def run_docker(
     """
     if keyfile is None:
         keyfile = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS", None)
-    filesystem.get_fs(outdir).makedirs(outdir, exist_ok=True)
-    with tempfile.NamedTemporaryFile(suffix=".yaml") as config_tempfile:
-        bind_mount_args = []
-        python_args = []
-        docker_args = []
-        config_location = _get_config_args(
-            config_dict_or_location, config_tempfile, bind_mount_args
-        )
-        _get_docker_args(docker_args, bind_mount_args, outdir)
-        _get_credentials_args(keyfile, docker_args, bind_mount_args)
-        _get_runfile_args(runfile, bind_mount_args, python_args)
-        python_command = _get_python_command(config_location, DOCKER_OUTDIR)
-        subprocess.check_call(
-            DOCKER_COMMAND
-            + bind_mount_args
-            + docker_args
-            + [docker_image]
-            + python_command
-            + python_args
-        )
-
-
-def _get_runfile_args(runfile, bind_mount_args, python_args):
-    if runfile is not None:
-        if filesystem._is_local_path(runfile):
-            bind_mount_args += ["-v", f"{os.path.abspath(runfile)}:{DOCKER_RUNFILE}"]
-            python_args += ["--runfile", DOCKER_RUNFILE]
-        else:
-            python_args += ["--runfile", runfile]
 
+    if isinstance(config_dict_or_location, str):
+        with fsspec.open(config_dict_or_location) as f:
+            config_dict = yaml.safe_load(f.read())
+    else:
+        config_dict = config_dict_or_location
 
-def _get_python_command(config_location, outdir):
-    return ["python3", "-m", FV3RUN_MODULE, config_location, outdir]
+    filesystem.get_fs(outdir).makedirs(outdir, exist_ok=True)
+    bind_mount_args = []
+    docker_args = []
+    _get_credentials_args(keyfile, docker_args, bind_mount_args)
+    _get_local_data_bind_mounts(config_dict, bind_mount_args)
 
+    outdir_in_docker = _get_docker_args(docker_args, bind_mount_args, outdir)
```

Comment:
should just use DOCKER_OUTDIR in place of this local variable (which will currently always be equal to DOCKER_OUTDIR) wherever you would use it.